libs/parallelism/algorithms/include/hpx/parallel/algorithms/uninitialized_move.hpp

```diff
@@ -36,64 +225,66 @@ namespace hpx { namespace parallel { inline namespace v1 {
     namespace detail {
         /// \cond NOINTERNAL
 
-        // provide our own implementation of std::uninitialized_move as some
-        // versions of MSVC horribly fail at compiling it for some types T
-        template <typename InIter1, typename InIter2>
-        InIter2 std_uninitialized_move(
-            InIter1 first, InIter1 last, InIter2 d_first)
+        ///////////////////////////////////////////////////////////////////////
+        template <typename InIter1, typename FwdIter2, typename Cond>
+        util::in_out_result<InIter1, FwdIter2> sequential_uninitialized_move(
+            InIter1 first, FwdIter2 dest, Cond cond)
         {
-            typedef
-                typename std::iterator_traits<InIter2>::value_type value_type;
+            using value_type =
+                typename std::iterator_traits<FwdIter2>::value_type;
 
-            InIter2 current = d_first;
+            FwdIter2 current = dest;
             try
             {
-                for (/* */; first != last; ++first, (void) ++current)
+                for (/* */; cond(first, current); (void) ++first, ++current)
```

Comment:
Could you use `HPX_INVOKE(cond, first, current)` instead, please?

Reply:
Done, thanks!